src/Compilers/Core/Portable/CryptographicHashProvider.cs

```diff
@@ -213,6 +216,60 @@ internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<byte> byte
             }
         }
 
+        static readonly byte[] _singleZeroByteArray = new byte[1] { 0 };
+
+        internal static ImmutableArray<byte> ComputeSourceHash(ImmutableArray<ConstantValue> constants, SourceHashAlgorithm hashAlgorithm = SourceHashAlgorithms.Default)
+        {
+            var algorithmName = GetAlgorithmName(hashAlgorithm);
+            using var incrementalHash = IncrementalHash.CreateHash(algorithmName);
+
+            foreach (var constant in constants)
+            {
+                incrementalHash.AppendData(getBytes(constant));
+            }
+
+            return ImmutableArray.Create(incrementalHash.GetHashAndReset());
+
+            static byte[] getBytes(ConstantValue constant)
+            {
+                Debug.Assert(Enum.GetValues(typeof(ConstantValueTypeDiscriminator)).Cast<ConstantValueTypeDiscriminator>().Max()
+                    == ConstantValueTypeDiscriminator.DateTime);
```

Comment:
It doesn't look like we depend on this condition below. If we are not handling a case, the code is going to throw below regardless underlying value of ```ConstantValueTypeDiscriminator.DateTime```. Instead, it might be better to add a comment in ```ConstantValueTypeDiscriminator``` that new values are likely to need special treatment in this function.

Reply:
Sure, I'll shift to a comment. 
That said this assertion would fail for existing tests as soon as a new `ConstantValueTypeDiscriminator` enum entry was added, before the right test was crafted. That would have also raised the right alarm.